src/python/pants/rules/core/cloc_test.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Sequence
+
+from pants.backend.jvm.targets.java_library import JavaLibrary
+from pants.build_graph.build_file_aliases import BuildFileAliases
+from pants.rules.core import cloc
+from pants_test.console_rule_test_base import ConsoleRuleTestBase
+
+
+class ClocTest(ConsoleRuleTestBase):
+  goal_cls = cloc.CountLinesOfCode
+
+  @classmethod
+  def rules(cls):
+    return super().rules() + cloc.rules()
+
+  @classmethod
+  def alias_groups(cls):
+    return BuildFileAliases(targets={ 'java_library': JavaLibrary })
+
+  def assert_counts(self, result: Sequence[str], lang: str, num_files: int, blank: int, comment: int, code: int):
+    for line in result:
+      fields = line.split()
+      if len(fields) >= 5:
+        if fields[0] == lang:
```

Comment:
Because you don't do anything with the else branch, these two conditionals can be combined to reduce the indentation:

```python
if len(fields) >= 5 and fields[0] == lang:
```

Even better imo is to invert the conditional for an early return. See https://www.refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html. This reduces the indentation even more and makes the intent a bit clearer imo that the main logic here is those assertions, but if certain conditions aren't first met then move on.

```python
for line in result:
  ...
  if len(fields) < 5 or fields[0] != lang:
    continue
  self.assertEqual(...)
  ...
  return
```

Reply:
(I know you copied this code - but imo worth the possible refactoring because we're unlikely to touch this code soon again)